webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java

```diff
@@ -103,48 +104,29 @@
-    public Map<String, List<Map<String, Object>>> getFavoriteSitesWithPages(@PathVariable String userId) throws UserNotDefinedException {
-
-        Session session = checkSakaiSession();
-        String requestUserId = session.getUserId();
-        //If non admin user requests data of another user, return empty map and warn
-        if(!securityService.isSuperUser() && !userId.equals(requestUserId)) {
-            log.warn("User " + requestUserId + " requested favorites from user " + userId);
-            return new HashMap<>();
-        }
-        return getUserFavoriteSitesWithPagesMap(userId);
-
-    }
-
-    @GetMapping(value = "/users/favorites", produces = MediaType.APPLICATION_JSON_VALUE) 
-    public Map<String, List<Map<String, Object>>> getFavoriteSitesWithPages() throws UserNotDefinedException {
-
-        Session session = checkSakaiSession();
-        String requestUserId = session.getUserId();
-        return getUserFavoriteSitesWithPagesMap(requestUserId);
+    @GetMapping(value = "/users/current/favourites", produces = MediaType.APPLICATION_JSON_VALUE) 
+    public Map<String, List<Map<String, Object>>> getFavouriteSitesWithPages() throws UserNotDefinedException {
 
-    }
+        String userId = checkSakaiSession().getUserId();
 
-    private Map<String, List<Map<String, Object>>> getUserFavoriteSitesWithPagesMap(String userId) throws UserNotDefinedException  {
+        ResourceProperties resourceProperties = preferencesService.getPreferences(userId)
+            .getProperties(PreferencesService.SITENAV_PREFS_KEY);
+        List<String> siteIds = resourceProperties.getPropertyList("order");
 
-        ResourceProperties resourceProperties = preferencesService.getPreferences(userId).getProperties(org.sakaiproject.user.api.PreferencesService.SITENAV_PREFS_KEY);
-        List<String> propertyList = resourceProperties.getPropertyList("order");
+        if (siteIds == null) return Collections.emptyMap();
 
-        if(propertyList == null) return new HashMap<>();
-
-        List<Map<String, Object>> sitesList = propertyList.stream().map(favoriteSiteId -> {
+        List<Map<String, Object>> sitesList = siteIds.stream().map(siteId -> {
             Map<String, Object> siteMap = new HashMap<>();
             try {
-                Site site = siteService.getSite(favoriteSiteId);
+                Site site = siteService.getSite(siteId);
                 siteMap.put("id", site.getId());
                 siteMap.put("title", site.getTitle());
                 siteMap.put("url", site.getUrl());
                 siteMap.put("type", site.getType());
                 List<Map<String, Object>> pageList = site.getOrderedPages().stream().map(page -> {
                     Map<String, Object> pageMap = new HashMap<>();
                     List<ToolConfiguration> toolList = page.getTools();
-                    if(toolList != null && toolList.size() == 1 ) {
+                    if (toolList.size() == 1 ) {
```

Comment:
Is it safe to remove the != null condition?

Reply:
Yes.